Framework/PythonInterface/mantid/FunctionWrapper.py

```diff
@@ -0,0 +1,258 @@
+from mantid.simpleapi import FunctionFactory
+
+class FunctionWrapper:
+# Wrapper class for Fitting Function 
+  def __init__ (self, name, **kwargs):
+    if not isinstance(name, str):
+       self.fun = name
+    else:
+       self.fun = FunctionFactory.createFunction(name)
+    
+       # Deal with attributes first
+#       for key in kwargs:
+#          if(self.fun.hasAttribute(key)):
+#              self.fun.storeAttributeValue(key, kwargs[key])
```

Comment:
This is because you said a bug prevented attributes from working. If fixed, the code could be restored.